server/mergin/sync/permissions.py

```diff
@@ -239,3 +239,27 @@ def get_user_project_role(project: Project, user: User) -> Optional[str]:
         return "writer"
     if ProjectPermissions.Read.check(project, user):
         return "reader"
+
+
+def project_admin_permissions(workspace_id, user: User) -> bool:
+    """Check if the user has admin permissions for workspace or a project in the workspace
+    True if user is either workspace admin or owner of a project in the workspace, False otherwise
+
+    :param workspace_id: workspace id
+    :type workspace_id: int
+    :param user: user
+    :type user: User
+    :rtype bool
+    """
+    if user.is_anonymous or not user.active:
+        return False
+    workspace = current_app.ws_handler.get(workspace_id)
+    if not workspace:
+        return False
+    if workspace.user_has_permissions(current_user, "admin"):
+        return True
+    projects = Project.query.filter_by(workspace_id=workspace.id).all()
```

Comment:
can we check this within single query instead of getting all projects just to check permissions on python side?